### PR TITLE
[FLINK-15205][hive][document] Update hive partition and orc for read_write_hive

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

read_write_hive is out of date, we need update it.

## Brief change log

- Add static and dynamic partition inserting
- Add orc vectorization optimization

## Verifying this change

document

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by JingsongLi** [comment]
Consider our literal is Decimal, we can not insert a table with double column by decimal, so change to int instead of double.

**Comment by JingsongLi** [comment]
Insert partition can not work with values, I create JIRA to better error message: https://issues.apache.org/jira/browse/FLINK-15272

File: docs/dev/table/hive/read_write_hive.md [hunk 1]
```diff
@@ -111,16 +111,44 @@ __________ __________
...
-Similarly, data can be written into hive using an `INSERT INTO` clause. 
+Similarly, data can be written into hive using an `INSERT` clause. Consider there is a mytable table with two columns: name, age.
 
 {% highlight bash %}
-Flink SQL> INSERT INTO mytable (name, value) VALUES ('Tom', 4.72);
+# ------ Insert with append mode ------ 
+Flink SQL> INSERT INTO mytable SELECT 'Tom', 25;
+
+# ------ Insert with overwrite mode ------ 
+Flink SQL> INSERT OVERWRITE mytable SELECT 'Tom', 25;
```
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]: are "append mode" and "overwrite mode" official names? 

maybe just say "insert new values by appending to existing table" and "insert new values by overwriting the whole existing table"

File: docs/dev/table/hive/read_write_hive.md [hunk 2]
```diff
@@ -111,16 +111,44 @@ __________ __________
...
+{% endhighlight %}
+
+We support partition table too, Consider there is a myparttable table with four columns: name, age, my_type and my_date. Column my_type and column my_date are the partition columns.
```
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]:
```suggestion
We support partitioned table too, Consider there is a partitioned table named myparttable with four columns: name, age, my_type and my_date, in types ...... my_type and my_date are the partition keys.
```

File: docs/dev/table/hive/read_write_hive.md [hunk 3]
```diff
@@ -111,16 +111,44 @@ __________ __________
...
+
+# ------ Insert with static(my_type) and dynamic(my_date) partition ------ 
+Flink SQL> INSERT OVERWRITE myparttable PARTITION (my_type='type_1') SELECT 'Tom', 25, '2019-08-08';
 {% endhighlight %}
 
 ## Formats
 
 We have tested on the following of table storage formats: text, csv, SequenceFile, ORC, and Parquet.
 
+# ------ ORC Vectorized Optimization ------
```
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]: is this turned on by default?

File: docs/dev/table/hive/read_write_hive.md [hunk 4]
```diff
@@ -111,16 +111,44 @@ __________ __________
...
 
 ## Formats
 
 We have tested on the following of table storage formats: text, csv, SequenceFile, ORC, and Parquet.
 
+# ------ ORC Vectorized Optimization ------ 
+Optimization is used automatically when the following conditions are met:
+
+- Columns without complex data type, like hive types: List, Map, Struct.
+- Hive version greater than or equal to version 2.
```
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]: version 2.0.0?

File: docs/dev/table/hive/read_write_hive.md [hunk 5]
```diff
@@ -111,16 +111,44 @@ __________ __________
...
 {% endhighlight %}
 
 ## Formats
 
 We have tested on the following of table storage formats: text, csv, SequenceFile, ORC, and Parquet.
 
+# ------ ORC Vectorized Optimization ------ 
+Optimization is used automatically when the following conditions are met:
+
+- Columns without complex data type, like hive types: List, Map, Struct.
```
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]: maybe "union" too?

File: docs/dev/table/hive/read_write_hive.md (hunk 1 above)
Comment (JingsongLi) [reply docs/dev/table/hive/read_write_hive.md]: Will modify to:
```
INSERT INTO will append to the table or partition, keeping the existing data intact.
INSERT OVERWRITE will overwrite any existing data in the table or partition.
```

File: docs/dev/table/hive/read_write_hive.md (hunk 3 above)
Comment (JingsongLi) [reply docs/dev/table/hive/read_write_hive.md]: Yes

Comment (bowenli86) [reply docs/dev/table/hive/read_write_hive.md]: can we document that?

**Pull request closed without merging**